src/Symfony/Component/HttpFoundation/BinaryFileResponse.php

```diff
@@ -157,6 +157,20 @@ public function setContentDisposition($disposition, $filename = '', $filenameFal
             $filename = $this->file->getFilename();
         }
 
+        if ('' === $filenameFallback && (!preg_match('/^[\x20-\x7e]*$/', $filename) || false !== strpos($filename, '%'))) {
+            $encoding = mb_detect_encoding($filename);
```

Comment:
`mb_detect_encoding($string, null, true)` to enable strict mode (non-strict is useless)